tpmutil/encoding.go

```diff
@@ -23,32 +23,49 @@ import (
 	"reflect"
 )
 
-// lengthPrefixSize is the size in bytes of length prefix for byte slices.
-//
-// In TPM 1.2 this is 4 bytes.
-// In TPM 2.0 this is 2 bytes.
-var lengthPrefixSize int
+// Encoding implements encoding logic for different versions of the TPM
+// specification.
+type Encoding struct {
+	// lengthPrefixSize is the size in bytes of length prefix for byte slices.
+	//
+	// In TPM 1.2 this is 4 bytes.
+	// In TPM 2.0 this is 2 bytes.
+	lengthPrefixSize int
+}
+
+var (
+	// TPM implements TPM 1.2 encoding.
+	TPM = &Encoding{
+		lengthPrefixSize: tpm12PrefixSize,
+	}
+	// TPM2 implements TPM 2.0 encoding.
+	TPM2 = &Encoding{
+		lengthPrefixSize: tpm20PrefixSize,
+	}
+
+	defaultEncoding *Encoding
+)
 
 const (
 	tpm12PrefixSize = 4
 	tpm20PrefixSize = 2
 )
 
-// UseTPM12LengthPrefixSize makes Pack/Unpack use TPM 1.2 encoding for byte
-// arrays.
+// UseTPM12LengthPrefixSize makes the package level Pack/Unpack functions use
+// TPM 1.2 encoding for byte arrays.
 func UseTPM12LengthPrefixSize() {
```

Comment:
Rename to `UseTPM12Encoding`

Reply:
If we're okay with backward incompatible changes in tpmutil, should we just remove the global encoder?